### PR TITLE
Trigger for ensuring only the newest state is stored #21

### PR DESCRIPTION
**Pull request by janekolszak**

### CODE REVIEW

File: src/db/nodeDb.js [hunk 1]
```diff
@@ -122,7 +134,7 @@ module.exports = {
       error_messages: readResult.cachedValue.errorMessages
     };
 
-    await nodeDb('states').insert(entry).onConflict(['contract_tx_id', 'sort_key']).ignore();
+    await nodeDb('states').insert(entry).onConflict(['contract_tx_id']).merge();
```
Comment (ppedziwiatr) [inline src/db/nodeDb.js:137]: will the trigger be also..triggered on merge?

File: src/db/00002_cleanup_indices.sql [hunk 2]
```diff
@@ -0,0 +1,12 @@
+BEGIN;
+
+-- Remove automatically created indices 
+DROP INDEX IF EXISTS states_sort_key_index;
+
+DROP INDEX IF EXISTS black_list_contract_tx_id_index;
```
Comment (ppedziwiatr) [inline src/db/00002_cleanup_indices.sql:6]: why are you removing this index? and the one for the errors?

Comment (janekolszak) [reply src/db/00002_cleanup_indices.sql:6]: Just a cleanup

File: src/db/nodeDb.js (hunk 1 above)
Comment (janekolszak) [reply src/db/nodeDb.js:137]: yes

**Pull request merged**